stdlib/3/os/__init__.pyi

```diff
@@ -285,63 +356,133 @@ def fstat(fd: int) -> stat_result: ...
 def fstatvfs(fd: int) -> statvfs_result: ...  # Unix only
 def fsync(fd: int) -> None: ...
 def ftruncate(fd: int, length: int) -> None: ...  # Unix only
+if sys.version_info >= (3, 5):
+    def get_blocking(fd: int) -> bool: ...  # Unix only
+    def set_blocking(fd: int, blocking: bool) -> None: ...  # Unix only
 def isatty(fd: int) -> bool: ...  # Unix only
+if sys.version_info >= (3, 3):
+    def lockf(fd: int, cmd: int, length: int) -> None: ...  # Unix only
 def lseek(fd: int, pos: int, how: int) -> int: ...
-def open(file: _PathType, flags: int, mode: int = ...) -> int: ...
+if sys.version_info >= (3, 3):
+    def open(file: _PathType, flags: int, mode: Optional[int] = ..., *, dir_fd: Optional[int] = ...) -> int: ...
+else:
+    def open(file: _PathType, flags: int, mode: Optional[int] = ...) -> int: ...
 def openpty() -> Tuple[int, int]: ...  # some flavors of Unix
 def pipe() -> Tuple[int, int]: ...
+if sys.version_info >= (3, 3):
+    def pipe2(flags: int) -> Tuple[int, int]: ...  # some flavors of Unix
+    def posix_fallocate(fd: int, offset: int, length: int) -> None: ...  # Unix only
+    def posix_fadvise(fd: int, offset: int, length: int, advice: int) -> None: ...  # Unix only
+    def pread(fd: int, buffersize: int, offset: int) -> bytes: ...  # Unix only
+    def pwrite(fd: int, string: bytes, offset: int) -> int: ...  # Unix only
 def read(fd: int, n: int) -> bytes: ...
+if sys.version_info >= (3, 3):
+    # sendfile() has a different variation on FreeBSD and Mac OS X
+    def sendfile(out_fd: int, in_fd: int, offset: Optional[int], count: int) -> int: ...  # Unix only
+    def readv(fd: int, buffers: Sequence[bytearray]) -> int: ...  # Unix only
+    def writev(fd: int, buffers: Sequence[bytes]) -> int: ...  # Unix only
+
+    terminal_size = NamedTuple('terminal_size', [('columns', int), ('lines', int)])
+    def get_terminal_size(fd: Optional[int] = ...) -> terminal_size: ...
```

Comment:
```
In [9]: os.get_terminal_size(None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-46d26e904443> in <module>()
----> 1 os.get_terminal_size(None)

TypeError: an integer is required (got type NoneType)
```
So it's not Optional.

Reply:
Another place where I was overzealous with Optionals. I think I fixed them all.